### PR TITLE
[Spark] Make naming of manage commit consistent

### PR DESCRIPTION
**Pull request by dhruvarya-db**
#### Which Delta project/connector is this regarding?

- [X] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

1. Replaces instances of Managed Commits with Managed Commit (without the s) in functions, configurations, and docs.
2. Renames the feature name to managedCommit-preview
3. Renames the DynamoDBCommitOwner tableConfig key to dynamoDBTableName

## How was this patch tested?

Existing tests should cover this refactor.

## Does this PR introduce _any_ user-facing changes?

The managed commit table feature name has been updated to managedCommit-preview

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala [hunk 1]
```diff
@@ -551,7 +551,7 @@ trait DeltaSQLConfBase {
   /////////////////////////////////////////////
 
   val MANAGED_COMMIT_DDB_AWS_CREDENTIALS_PROVIDER_NAME =
-    buildConf("managedCommits.commitOwner.ddb.awsCredentialsProviderName")
+    buildConf("managedCommit.commitOwner.ddb.awsCredentialsProviderName")
```
Comment (prakharjain09) [inline spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala:554]: Could we make it more explicit: `managedCommit.dynamoDBCommitOwner.awsCredentialsProviderName` or `managedCommit.dynamoDbCommitOwner.awsCredentialsProviderName`?

File: spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala [hunk 2]
```diff
@@ -560,7 +560,7 @@ trait DeltaSQLConfBase {
       .createWithDefault("com.amazonaws.auth.DefaultAWSCredentialsProviderChain")
 
   val MANAGED_COMMIT_DDB_SKIP_PATH_CHECK =
-    buildConf("managedCommits.commitOwner.ddb.skipPathCheck.enabled")
+    buildConf("managedCommit.commitOwner.ddb.skipPathCheck.enabled")
```
Comment (prakharjain09) [inline spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala:563]:
```suggestion
    skipPathCheckEnabled
```

**Review by prakharjain09 — APPROVED** [review]
Thanks for making the naming consistent!!

File: spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala (hunk 1 above)
Comment (dhruvarya-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala:554]: How about managedCommit.commitOwner.dynamodb.awsCredentialsProviderName

**Pull request merged**